### PR TITLE
fix: remove PR if max attempt reached

### PR DESCRIPTION
**Pull request by m-rudyk**
# Pull Request

## Description

Please include a summary of the change and which issue is fixed. Also, provide relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [ ] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [ ] Test Coverage update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**Test Configuration**:
* Java: v17
* LPVS Release: v1.x.x

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code meets the required code coverage for lines (90% and above)
- [ ] My code meets the required code coverage for branches (80% and above)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 1]
```diff
@@ -271,7 +271,24 @@ public void processWebHook(LPVSQueue webhookConfig) {
             pullRequest = lpvsPullRequestRepository.saveAndFlush(pullRequest);
             log.error("Can't authorize commentResults() " + e);
             e.printStackTrace();
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java]: We don't need to print stack trace. Please remove this line.

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 2]
```diff
@@ -271,7 +271,24 @@ public void processWebHook(LPVSQueue webhookConfig) {
...
-            delete(webhookConfig);
+            int currentAttempts = webhookConfig.getAttempts();
+            if (currentAttempts < maxAttempts) {
+                webhookConfig.setAttempts(currentAttempts++);
+                try {
+                    addFirst(webhookConfig);
+                } catch (InterruptedException e1) {
+                    log.warn("Failed to update Queue element");
+                    e1.printStackTrace();
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java]: We don't need to print stack trace. Please remove this line.
You can take just exception message for debug purpose.

File: src/main/java/com/lpvs/service/LPVSQueueService.java [hunk 3]
```diff
@@ -271,7 +271,24 @@ public void processWebHook(LPVSQueue webhookConfig) {
...
+                }
+                queueRepository.save(webhookConfig);
+            } else {
+                log.warn(
+                        "Maximum amount of processing webhook reached for pull request: "
+                                + pullRequest.getId()
+                                + " "
+                                + pullRequest.toString());
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/service/LPVSQueueService.java]: This line is strange. We have no custom toString() for pullRequest. 
It's better to print some meaningful info or implement custom toString().

**Review by o-kopysov — COMMENTED** [review]
Overall logic looks good, but I need to debug more.
I left several comments that require small fixes. Please check.

File: src/main/java/com/lpvs/service/LPVSQueueService.java (hunk 3 above)
Comment (m-rudyk) [reply src/main/java/com/lpvs/service/LPVSQueueService.java]: I've replaced to pull request URL for easy access not by ID.

**Pull request closed without merging**